### DISCUSSION TITLE
How do i utilise torch.load() for my saved GPU model: Seeking Guidance

### DISCUSSION THREAD

**Discussion by AquifersBSIM** [chemprop/chemprop#948] [Q&A]
As the title goes, I have a GPU model and I would want to use that model to predict some target_values. I am also running on an CPU-only machine. The only documentation I found are: https://chemprop.readthedocs.io/en/main/autoapi/chemprop/models/index.html#chemprop.models.MPNN.load_from_checkpoint. FYI: I am quite new to this!

This is what i used:
```
map_location=torch.device('cpu')
mpnn = models.MPNN.load_from_checkpoint(checkpoint_input, map_location=map_location)
```
The Runtime error I got:
```
RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.
```

**Comment by KnathanM** [comment]
Hi @AquifersBSIM, sorry I didn't see your post. Did you get your issue resolved?

**Reply by AquifersBSIM** [reply]
Hi @KnathanM, same here lol. FWIW, I believe my issue is resolved by just setting the accelerator to be "auto" (for predictions):

```python
trainer = pl.Trainer(
                    logger=None,
                    enable_progress_bar=True,
                    accelerator="auto",
                    devices=1
                )
```

But I have a question though, why is the GPU usage for, when training (accelerator='gpu') is always low, say 10 to 20%? I remember setting the batch size to be crazily huge, just to see if GPU usage would increase, but to no avail.

**Reply by KnathanM** [reply]
Could have to do with the number of workers for featurizing the graph on the CPU before sending the batch to the GPU. https://github.com/Lightning-AI/pytorch-lightning/discussions/14207
You could try increasing the number of workers when making the dataloader or try caching all the featurized graph at the beginning https://chemprop.readthedocs.io/en/latest/tutorial/python/data/datasets.html#Caching

**Reply by JacksonBurns** [reply]
Also see this discussion about why GPU usage can remain low: https://github.com/chemprop/chemprop/discussions/1019
